### ISSUE TITLE
Sync Issues (v. 6.2.5)

### ISSUE THREAD

**Issue by Doc73** [mozilla/multi-account-containers#1708]
- Multi-Account Containers Version: 6.2.5
- Operating System + Version: Debian 10 Buster
- Firefox Version: 75 & 68.7ESR
- Other installed Add-ons + Version + Enabled/Disabled-Status: 

- - Amazon.it	1.1	true	amazon@search.mozilla.org
- - Bing	1.1	true	bing@search.mozilla.org
- - DuckDuckGo	1.0	true	ddg@search.mozilla.org
- - eBay	1.0	true	ebay@search.mozilla.org
- - Firefox Multi-Account Containers	6.2.5	true	@testpilot-containers
- - FoxyProxy Standard	7.4.3	true	foxyproxy@eric.h.jung
- - Google	1.0	true	google@search.mozilla.org
- - Integrazione con GNOME Shell	10.1	true	chrome-gnome-shell@gnome.org
- - JabRef Browser Extension	2.2	true	@jabfox
- - Keepa - Amazon Price Tracker	3.48	true	amptra@keepa.com
- - KeePassXC-Browser	1.6.2	true	keepassxc-browser@keepassxc.org
- - Nano Adblocker	1.0.0.141	true	{af881826-e0f9-4b97-898a-a416dc473545}
- - Nano Defender for Firefox	15.0.0.197	true	{fcf60470-b210-4c17-969e-9ae01491071e}
- - Notes by Firefox	4.3.5	true	notes@mozilla.com
- - Privacy Badger	2020.2.19	true	jid1-MnnxcxisBPnSXQ@jetpack
- - Wikipedia (it)	1.0	true	wikipedia@search.mozilla.org
- - Worldwide Radio	3.4	true	worldwide@radio
- - Zotero Connector	5.0.64	true	zotero@chnm.gmu.edu
- - Facebook Container	2.1.0	false	@contain-facebook
- - Plasma Integration	1.7	false	plasma-browser-integration@kde.org

### Actual behavior
Latest version solves issue #1703 , but continues to prevent Firefox from syncing.

### Steps to reproduce
1. Install Firefox Multi-Account Containers
2. Try syncinc Firefox
3. :cry: 

Thank you very much in advance,
Domenico

**Comment by Wielding** [comment]
Same issue using Windows 10 and Ubuntu 19.04.  I set up containers on Windows while signed into Mozilla with sync enabled and then installed on Ubuntu while signed in to Mozilla with sync enabled and it does not load my supposedly synced container definitions.  Multi-Account containers version 6.2.5 and Firefox 75.0 on both machines.

**Comment by jacobvenborg** [comment]
This is same for me; I have 3 devices running Windows 10, FF 75 and MAC 6.2.5. I've set up multiple containers on one laptop, but the other devices won't sync at all. This was also the case prior FF 75 update, unfortunately.

**Comment by Wielding** [comment]
I was able to get it partially working by stopping "Add-ons" and "Options" syncing form my Mozilla account and then re-enabling them.  It brought in my containers from my other machine but left the default containers and did not sync the sites associated with each one.

**Comment by jacobvenborg** [comment]
Thanks for suggesting this! I've also tried this already. Did not work for me, unfortunately.

And sorry to hear it didn't sync the sites associations; syncing containers without associations seems kind of useless, lol.

**Comment by jagenheim** [comment]
Same problem.

Always running out of resources when it comes to syncing the MAC:
Sync.Engine.Extension-Storage	ERROR	Syncing @testpilot-containers: request failed: Error: HTTP 507 Insufficient Storage: Resource access is forbidden for this user (Maximum bytes per object exceeded " "(46287 > 16384 Bytes.)(resource://services-common/kinto-http-client.js:2771:5) JS Stack trace: ServerResponse@kinto-http-client.js:2771:5
processResponse@kinto-http-client.js:2925:13

This is consistent over 3 different computers. All that is changing is the byte values.

I have tried to disable all add-ons (except MAC) and syncing of tabs and history in case they are eating up all space that MAC needs, but still have the same problem...

**Comment by jagenheim** [comment]
After deleting the add-on 'Temporary Containers' and forceably deleting the records from sync using the 'about sync' add-on, I can now sync again.

**Comment by Doc73** [comment]
@jagenheim 
Yes, but I use neither "Temporary Containers" nor "about sync" addons... I have never used them, and indeed I did not even know them. :smile:

**Comment by jagenheim** [comment]
Has anyone figured out where it saves which sites go to which containers?

I can always manually copy the 'containers.json' to avoid the broken sync, but that does not contain the information which site are mapped to which container.

**Comment by Solid-Ice8** [comment]
I have been trying to find the specific location as well.

I read somewhere (either from Mozilla page itself or this add-on) that it's in a "Storage" location.

The page mentioned it's almost impossible to convert to readable language (for computers too).

**Comment by jasonekratz** [comment]
So I've noticed that it only seems to sync the containers after a restart of Firefox. I'm trying to sync between Mac and Windows and have definitely reproduced this. Add a container on Windows or Mac. Force a sync. Restart the other browser and sync. The new container appears. None of these edits seem to finally really sync without a browser restart.

**Comment by mcepl** [comment]
7.0.1

1. Borked my profile somehow
2. Created new profile, syncing with my Firefox Account (with Mozilla/5.0 (X11; Linux x86_64; rv:78.0) Gecko/20100101 Firefox/78.0)

Expected behaviour:
All websites open in the same containers as before

Observed behaviour:
All containers are lost, no webpage remembers where it is supposed to be.

Additional information:
Error logs from about:sync
[error-sync-1594979632785.txt](https://github.com/mozilla/multi-account-containers/files/4937096/error-sync-1594979632785.txt)
[error-sync-1594980154917.txt](https://github.com/mozilla/multi-account-containers/files/4937097/error-sync-1594980154917.txt)

**Comment by boistordu** [comment]
Maybe the title should be changed since it s a continuing problem.
V 7.1.0

Firefox 83
What should I look for in the error sync log? Which name of extension ?

It would sync only if I enable the addon option in the sync option of the Firefox account. 
But I would rather not have to do that ever again because my devices are vm and different hosts with different horsepower. So all addons shouldn't be the same on every of my hosts. So is this an requirement for this to be activated?

**Comment by jacob-lee** [comment]
Sync is not working for me.

**Comment by riccoho** [comment]
Deleted container respawn, removed site reappear... it's quite inconvenient

**Comment by AndrewTerry** [comment]
This problem still exists. I'm setting up a new laptop with Firefox 134.0.1 (which comes with Ubuntu). I've enabled Sync in the browser itself, and checked the `Enable syncronization` box in the Multi-Account Containers add-on.

The source and target machines have had Firefox open and running for some time, and while Bookmarks, Add-Ons, Open Tabs, etc have all synced correctly, the Container sites have still have not. Are there any diagnostics/logging I can enable to see what's going on?
